crates/uv/src/commands/project/run.rs

```diff
@@ -1071,6 +1071,71 @@ hint: If you are running a script with `{}` in the shebang, you may need to incl
                 requirements_site_packages.escape_for_python(),
             ))?;
 
+            // N.B. The order here matters — earlier interpreters take precedence over the
+            // later ones.
+            for interpreter in [requirements_env.interpreter(), &base_interpreter] {
+                // Copy each entrypoint from the base environments to the ephemeral environment.
+                for entry in fs_err::read_dir(interpreter.scripts())? {
+                    let entry = entry?;
+
+                    if !entry.file_type()?.is_file() {
+                        continue;
+                    }
+
+                    copy_entrypoint(
+                        &entry.path(),
+                        &ephemeral_env.scripts().join(entry.file_name()),
+                        interpreter.sys_executable(),
+                        ephemeral_env.sys_executable(),
+                    )?;
+                }
+
+                // Link common data directories from the base environment to the ephemeral
+                // environment. This is critical for Jupyter Lab, which cannot operate without the
+                // files it writes to `<prefix>/share`.
+                //
+                // We only perform a shallow merge here, so `<prefix>/share/<name>` will only be
+                // written once and contents beyond the first level, e.g.,
+                // `<prefix>/share/<name>/foo` will not be merged across parent interpreters.
+                for dir in &["etc", "share"] {
```

Comment:
Since this is just for Jupyter right now, we could narrow this to just apply to the `share/jupyter` directory if we wanted 🤔

Reply:
I narrowed this